### DISCUSSION TITLE
include package names in "of type 'Apple' to 'Apple'" error messages

### DISCUSSION THREAD

**Discussion by jrfondren** [odin-lang/Odin#2146] [Ideas/Requests]
Consider the following program:

```odin
package main

import "core:fmt"
import "core:bytes"

Buffer :: struct {}

main :: proc() {
	b1 : [^]bytes.Buffer = &bytes.Buffer{}
	b2 := Buffer{}
	p1 : [^]Buffer = &b2
	b1 = p1
}
```
This fails to compile with this error message:
```
Cannot assign value 'p1' of type '[^]Buffer' to '[^]Buffer' in assignment
```

Here it's clear what the problem is, but it's frequently not clear, and has come up in beginner discussions where a custom `Buffer` was accidentally passed to a procedure expecting a Vulkan `Buffer`, and instead of the error message immediately revealing that problem, it looked like a problem the programmer's understanding of pointers, or like a bug in the language.

### Alternatives

```
Cannot assign value 'p1' of type '[^]Buffer (main)' to '[^]Buffer (bytes)' in assignment

Cannot assign value 'p1' of type '[^]main.Buffer' to '[^]bytes.Buffer' in assignment
```

To *always* include the package information might just make for noisier error messages, but it should definitely be included when it would disambiguate types like this.

### Prior references

This came up and was mentioned as a specific enhancement to be made in #83 , but there was a more serious bug going on with that issue as the packages in an improved error message would've been the same.

**Comment by Kelimion** [comment]
It would only need to add the `package.` prefix or ` (package)` suffix if the type names are otherwise identical. That would avoid the noise.
